Xamarin.Forms.Platform.Android/ImageCache.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Threading.Tasks;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	/// <summary>
+	/// I setup the access to all the cache elements to be async because
+	/// if I didn't then it was locking up the GC and freezing the entire app
+	/// </summary>
+	class ImageCache
+	{
+
+		const string _cachePlaceHolder = "PLEASEHOLD";
+
+		global::Android.Util.LruCache _lruCache;
+
+		static int GetCacheSize()
+		{
+			// https://developer.android.com/topic/performance/graphics/cache-bitmap
+			int cacheSize = 4 * 1024 * 1024;
+			if (Java.Lang.Runtime.GetRuntime()?.MaxMemory() != null)
+			{
+				var maxMemory = (int)(Java.Lang.Runtime.GetRuntime().MaxMemory() / 1024);
+				cacheSize = maxMemory / 8;
+			}
+			return cacheSize;
+		}
+
+		public ImageCache() : base()
+		{
+			_lruCache = new global::Android.Util.LruCache(GetCacheSize());
```

Comment:
So a note here...

`LruCache` seems like it functions on the "number of entries": https://developer.android.com/reference/android/util/LruCache#LruCache(int)

_Not_ the size in bytes... That means the pool will only hold N entries of images--regardless of those images' sizes.

When I implemented something like this myself, I actually ported `LruCache` to C# and changed it to take the actual size of the `Bitmap` into account by using [Bitmap.ByteCount](https://developer.android.com/reference/android/graphics/Bitmap#getByteCount()).